File: spectra-starter/spectra-starter-common/src/main/java/com/yangxj96/spectra/starter/common/converter/StringToLocalDateTimeConverter.java
package com.yangxj96.spectra.starter.common.converter;

import com.yangxj96.spectra.starter.common.props.JacksonProperties;
import jakarta.annotation.Resource;
import org.springframework.core.convert.converter.Converter;
import org.springframework.lang.Nullable;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * 字符串转LocalDatetime的Converter
 */
public class StringToLocalDateTimeConverter implements Converter<String, LocalDateTime> {

    @Resource
    private JacksonProperties props;

    @Override
    @Nullable
    public LocalDateTime convert(String source) {
        return LocalDateTime.parse(source.trim(), DateTimeFormatter.ofPattern(props.getLocalDateTimeFormat()));
    }

}

File: yangxj96-starter/yangxj96-starter-common/src/main/kotlin/com/yangxj96/saas/starter/common/converter/StringToLocalDateTimeConverter.kt
package com.yangxj96.saas.starter.common.converter

import com.yangxj96.saas.starter.common.props.JacksonProperties
import jakarta.annotation.Resource
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter


/**
 * 字符串转LocalDatetime的Converter
 */
class StringToLocalDateTimeConverter : Converter<String, LocalDateTime> {

    @Resource
    private lateinit var props: JacksonProperties

    override fun convert(source: String): LocalDateTime? {
        return LocalDateTime.parse(source.trim { it <= ' ' }, DateTimeFormatter.ofPattern(props.localDateTimeFormat))
    }

}